crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -336,12 +347,34 @@ impl AlgorithmUpdaterV1 {
             max_change_percent: self.max_da_gas_price_change_percent,
 
             latest_da_cost_per_byte: self.latest_da_cost_per_byte,
-            total_rewards: self.total_da_rewards,
+            total_rewards: self.total_da_rewards_excess,
             total_costs: self.projected_total_da_cost,
             last_profit: self.last_profit,
             second_to_last_profit: self.second_to_last_profit,
             da_p_factor: self.da_p_component,
             da_d_factor: self.da_d_component,
         }
     }
+
+    // We only need to track the difference between the rewards and costs after we have true DA data
+    // Normalize, or zero out the lower value and subtract it from the higher value
+    fn normalize_rewards_and_costs(&mut self) {
+        let total_rewards = self.total_da_rewards_excess;
+        let total_costs = self.latest_known_total_da_cost_excess;
+        if total_rewards > total_costs {
+            let excess = total_rewards.saturating_sub(total_costs);
+            let projected_cost_excess =
+                self.projected_total_da_cost.saturating_sub(total_costs);
+            self.projected_total_da_cost = projected_cost_excess;
+            self.total_da_rewards_excess = excess;
+            self.latest_known_total_da_cost_excess = 0;
+        } else {
+            let excess = total_costs.saturating_sub(total_rewards);
+            let projected_cost_excess =
+                self.projected_total_da_cost.saturating_sub(total_rewards);
+            self.projected_total_da_cost = projected_cost_excess;
+            self.total_da_rewards_excess = 0;
+            self.latest_known_total_da_cost_excess = excess;
+        }
+    }
```

Comment:
```suggestion
  fn normalize_rewards_and_costs(&mut self) {
      let (excess, projected_cost_excess) = if self.total_da_rewards_excess > self.latest_known_total_da_cost_excess {
          (
              self.total_da_rewards_excess.saturating_sub(self.latest_known_total_da_cost_excess),
              self.projected_total_da_cost.saturating_sub(self.latest_known_total_da_cost_excess),
          )
      } else {
          (
              self.latest_known_total_da_cost_excess.saturating_sub(self.total_da_rewards_excess),
              self.projected_total_da_cost.saturating_sub(self.total_da_rewards_excess),
          )
      };
  
      self.projected_total_da_cost = projected_cost_excess;
      if self.total_da_rewards_excess > self.latest_known_total_da_cost_excess {
          self.total_da_rewards_excess = excess;
          self.latest_known_total_da_cost_excess = 0;
      } else {
          self.total_da_rewards_excess = 0;
          self.latest_known_total_da_cost_excess = excess;
      }
  }
```
extracting common logic out of the branches

Reply:
can we also use `checked_sub` here instead?

Reply:
Can you be more specific about why we'd prefer `checked_sub`?

Reply:
well, we should have an error if the value becomes negative, right?

Reply:
In theory it can't, so maybe it would be most correct to do an `expect` here with a message explaining why it'll never happen.

Reply:
I'd rather the function be infallible. Hopefully we can prove to ourselves that the algorithm can't get into that state with tests rather than adding `Result` here.

Reply:
okay, sounds good